### DISCUSSION TITLE
How to load stb library locally?(BUILD_DEPENDENCIES "LOCAL")

### DISCUSSION THREAD

**Discussion by layfolk233** [zxing-cpp/zxing-cpp#801] [Q&A]
Due to the network or other unknown reasons, when I execute cmake, I often can't compile it because I can't find the stb library. Only occasionally can cmake succeed by taking a few chances.

$ cmake ..
-- Checking for module 'stb'
--   No package 'stb' found
[ 11%] Performing download step (git clone) for 'stb-populate'
Cloning into 'stb-src'...
fatal: unable to access 'https://github.com/nothings/stb.git/': gnutls_handshake() failed: The TLS connection was non-properly terminated.
Cloning into 'stb-src'...
fatal: unable to access 'https://github.com/nothings/stb.git/': Empty reply from server
Cloning into 'stb-src'...
fatal: unable to access 'https://github.com/nothings/stb.git/': Failed to connect to github.com port 443: Connection refused
-- Had to git clone more than once:
          3 times.
CMake Error at stb-subbuild/stb-populate-prefix/tmp/stb-populate-gitclone.cmake:31 (message):
  Failed to clone repository: 'https://github.com/nothings/stb.git'

Then I noticed that there is an option in cmakelists.txt to load stb library in LOCAL mode, but I tried to download stb library (https://github.com/nothings/stb.git/) myself. I don't know what to do next. Can you help me? thank you
![1719646250431](https://github.com/zxing-cpp/zxing-cpp/assets/73208956/6b27189e-4f9e-4306-9063-fb811c5774ad)
![1719646341714](https://github.com/zxing-cpp/zxing-cpp/assets/73208956/58a8b767-8eb5-4763-b86f-acfb42bfc646)

**Comment by axxel** [comment]
You have to make sure the header files are in your include path, e.g.
```
cmake -DCMAKE_CXX_FLAGS=-I\ /path/to/include <path/to/source>
```
The easiest way to do this on Ubuntu is with actually with installing a package: `sudo apt install libstb-dev`.

**Comment by layfolk233** [comment]
Thank you, trying the two methods you mentioned still can't solve it, but then I thought that even if I realized the local loading of stb during cmake, there are other resource files that need to be loaded from github (such as fmt), which should still not pass cmake, so I tried to solve the network problem.

**Comment by bkmgit** [comment]
This was fixed in https://github.com/zxing-cpp/zxing-cpp/commit/764f7ac3438f0e7638da27ad00cc2147312a2ea6

When generating the configuration files use
```
cmake .. -DSTB_IMAGE_INCLUDE_DIR=PATH_TO_STB_IMAGE_HEADER
```
